types/src/v18/blockchain/mod.rs

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! The JSON-RPC API for Bitcoin Core `v0.18` - blockchain.
+//!
+//! Types for methods found under the `== Blockchain ==` section of the API docs.
+
+mod into;
+
+use serde::{Deserialize, Serialize};
+
+use super::{MempoolEntryError, MempoolEntryFees};
+
+/// Result of JSON-RPC method `getmempoolentry`.
+///
+/// > getmempoolentry txid
+/// >
+/// > Returns mempool data for given transaction
+/// >
+/// > Arguments:
+/// > 1. "txid"                 (string, required) The transaction id (must be in mempool)
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct GetMempoolEntry(pub MempoolEntry);
+
+/// A relative (ancestor or descendant) transaction of a transaction in the mempool.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct MempoolEntry {
+    /// Virtual transaction size as defined in BIP 141.
+    ///
+    /// This is different from actual serialized size for witness transactions as witness data is discounted.
+    pub size: i64,
+    /// DEPRECATED: Transaction fee in BTC.
+    pub fee: f64,
+    /// DEPRECATED: Transaction fee with fee deltas used for mining priority.
+    #[serde(rename = "modifiedfee")]
+    pub modified_fee: f64,
+    /// Local time transaction entered pool in seconds since 1 Jan 1970 GMT.
+    pub time: i64,
+    /// Block height when transaction entered pool.
+    pub height: i64,
+    /// Number of in-mempool descendant transactions (including this one).
+    #[serde(rename = "descendantcount")]
+    pub descendant_count: i64,
+    /// Virtual transaction size of in-mempool descendants (including this one).
+    #[serde(rename = "descendantsize")]
+    pub descendant_size: i64,
+    /// DEPRECATED: Modified fees (see above) of in-mempool descendants (including this one).
+    #[serde(rename = "descendantfees")]
+    pub descendant_fees: f64,
+    /// Number of in-mempool ancestor transactions (including this one).
+    #[serde(rename = "ancestorcount")]
+    pub ancestor_count: i64,
+    /// Virtual transaction size of in-mempool ancestors (including this one).
+    #[serde(rename = "ancestorsize")]
+    pub ancestor_size: i64,
+    /// DEPRECATED: Modified fees (see above) of in-mempool ancestors (including this one).
+    #[serde(rename = "ancestorfees")]
+    pub ancestor_fees: f64,
+    /// Hash of serialized transaction, including witness data.
+    pub wtxid: String,
+    /// (No docs in Core v0.17.)
```

Comment:
I merged but perhaps we want to follow up and change this to either make up some docs or remove the version number?

Reply:
I have redone this method for the changes in all version in #252 and added docs to the `fees` field.